### PR TITLE
add finalizers to legacy resource deletion magic

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/5806. Discussed with Mr Pawel. I still need to test this.

### CODE REVIEW

File: service/controller/v3/resource/legacy/resource.go [hunk 1]
```diff
@@ -265,18 +265,16 @@ func (r *Resource) newDeleteChange(ctx context.Context, obj, currentState, desir
 		_, err := r.k8sClient.CoreV1().Namespaces().Create(ns)
 		if apierrors.IsAlreadyExists(err) {
 			namespace, err := r.k8sClient.CoreV1().Namespaces().Get(ns.GetName(), metav1.GetOptions{})
-			if apierrors.IsNotFound(err) {
```
Comment (xh3b4sd) [inline service/controller/v3/resource/legacy/resource.go]: This does not make sense. The namespace cannot exist already AND not be found.

**Review by MarcelMue — APPROVED** [review]

File: service/controller/v3/resource/legacy/resource.go [hunk 2]
```diff
@@ -265,18 +265,16 @@ func (r *Resource) newDeleteChange(ctx context.Context, obj, currentState, desir
...
-				r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-
-				return nil, nil
-			} else if err != nil {
+			if err != nil {
 				return nil, microerror.Mask(err)
 			}
 
 			if namespace != nil && namespace.Status.Phase == "Terminating" {
 				r.logger.LogCtx(ctx, "level", "debug", "message", "consider network cleanup done")
```
Comment (kopiczko) [inline service/controller/v3/resource/legacy/resource.go]:
```suggestion
				r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("destroyer namespace is in phase %#q", namespace.Status.Phase))
```

**Review by kopiczko — APPROVED** [review]

**Comment by kopiczko** [comment]
Nit: the title should contain rather keep than add.

**Review by tuommaki — APPROVED** [review]

**Comment by xh3b4sd** [comment]
I tested cluster creation and deletion on `geckon` and it works. No leftover cluster role bindings anymore as it seems. PM sorted. Thanks Pawel for brainwashing me. <3

**Pull request merged**